### PR TITLE
Add input placeholder class

### PR DESCRIPTION
**Pull request by UnDarkle**

### CODE REVIEW

File: src/rotest/core/__init__.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
-from flow import TestFlow
-from case import TestCase
-from block import TestBlock
-from suite import TestSuite
-from abstract_test import request
-from flow_component import PipeTo, MODE_CRITICAL, MODE_FINALLY, MODE_OPTIONAL
+from flow import TestFlow
```
Comment (osherdp) [inline src/rotest/core/__init__.py]: Use correct relative imports:
```
from .flow import TestFlow
```

File: src/rotest/core/abstract_test.py [hunk 2]
```diff
@@ -1,244 +1,244 @@
...
+        resources_dict = {name: resource
+                          for name, resource in self.locked_resources.items()
+                          if name in resources}
+
+        self.resource_manager.release_resources(resources_dict,
+                                                dirty=dirty,
+                                                force_release=force_release)
+
+        # Remove the resources from the test's resource to avoid double release
+        for key in resources_dict.itervalues():
```
Comment (osherdp) [inline src/rotest/core/abstract_test.py]: Change to ```resource```

File: docs/advanced/blocks.rst [hunk 3]
```diff
@@ -1,363 +1,318 @@
...
+   multiprocess, TIMEOUT, etc. are available also for ``TestFlow`` class.
+
+TestBlock
+---------
+
+#. ``inputs``: define class fields and assign them to instances of BlockInput
+    to ask for values for the block (values are passed via ``common``,
+    ``parametrize``, previous blocks passing them as ``outputs``, or as
+    requested resources of the block or its containers).
+    You can pass a default value to BlockInput to assign if non is supplied
```
Comment (osherdp) [inline docs/advanced/blocks.rst]: pass->define

File: src/rotest/core/block.py [hunk 4]
```diff
@@ -1,126 +1,188 @@
...
+    Defines tests that are parts of a greater test. The block is dependent on
+    the containing :class:`rotest.core.flow.TestFlow` to lock and pass
+    resources to it.
+
+    Test blocks can only be skipped because of test related issues
+    (e.g. no resources, test-flow stopped due to failure, intentional skip)
+    and can't be skipped on account of 'run_delta' (passed in previous runs),
+    tags filtering, etc.
+
+    declaring 'inputs': assign class fields to instances of BlockInput to
```
Comment (osherdp) [inline src/rotest/core/block.py]: Capital D. also below

File: src/rotest/core/__init__.py (hunk 1 above)
Comment (UnDarkle) [reply src/rotest/core/__init__.py]: k

File: src/rotest/core/abstract_test.py (hunk 2 above)
Comment (UnDarkle) [reply src/rotest/core/abstract_test.py]: k

File: docs/advanced/blocks.rst (hunk 3 above)
Comment (UnDarkle) [reply docs/advanced/blocks.rst]: k

File: src/rotest/core/block.py (hunk 4 above)
Comment (UnDarkle) [reply src/rotest/core/block.py]: k

**Pull request merged**